### ISSUE TITLE
Propagating prefect task names to prefect tasks instead of it showing up as _run_prefect_task

### ISSUE THREAD

**Issue by achordia20** [PrefectHQ/prefect-ray#86]
# Expectation / Proposal
Propagate the prefect task name (whether function or manual name override in the task decorator) to the ray task.

# Traceback / Example
![Screen Shot 2023-05-18 at 12 43 20 PM](https://github.com/PrefectHQ/prefect-ray/assets/1892142/4d158129-ad26-4937-9cd9-abf3f96269d9)

- [ ] I would like to [help contribute](https://PrefectHQ.github.io/prefect-ray/#contributing) a pull request to resolve this!

**Comment by azmyrajab** [comment]
Curious if anyone has been able to take a look at this? 

One could achieve this by using .options as per below

```
        if remote_options:
            ray_decorator = ray.remote(**remote_options)
        else:
            ray_decorator = ray.remote

        self._ray_refs[key] = ray_decorator(self._run_prefect_task).options(name=user_specified_task_name).remote(
            sync_compatible(call.func), *upstream_ray_obj_refs, **call_kwargs
        )
```

**Comment by j-tr** [comment]
@azmyrajab 
Here's my attempt to solve this: #103. Happy to incorporate any feedback.
From what I understand, options can only be used to change properties that can be set via the remote options (i.e. parameters to https://docs.ray.io/en/latest/ray-core/api/doc/ray.remote.html). Name is none of them.

**Comment by azmyrajab** [comment]
Hi Justin, thank you so much for taking a look at this

Fyi its a trick that does work, but I think its a feature that is not well documented as you noticed. It was added here https://github.com/ray-project/ray/issues/10371 to address these type of scenarios where a factory function produces multiple tasks. 

Here is an example I just ran by 'patching' the Ray Prefect source:

<img width="549" alt="image" src="https://github.com/PrefectHQ/prefect-ray/assets/9899436/30632430-d639-40ca-99ec-205c40f965b5">

```
        if remote_options:
            ray_decorator = ray.remote(**remote_options)
        else:
            ray_decorator = ray.remote

        self._ray_refs[key] = ray_decorator(self._run_prefect_task).options(name="hello_world").remote(
            sync_compatible(call.func), *upstream_ray_obj_refs, **call_kwargs
        )
```

In any case taking a look at your https://github.com/PrefectHQ/prefect-ray/pull/103 now - will be very grateful if it solves this problem no matter

**Comment by azmyrajab** [comment]
Yes, I think the ray feature is intended for that, and instead of modifying __qualname__ you can propagate the prefect task name via options and I think this will be a nice solution

**Comment by j-tr** [comment]
thank you for pointing me to [#10371](https://github.com/ray-project/ray/issues/10371). just changed my MR to use options instead.

**Comment by j-tr** [comment]
[desertaxle](https://github.com/desertaxle) happy to receive any feedback on #103.
